packages/redux/src/raffles/reducer.ts

```diff
@@ -0,0 +1,180 @@
+import * as actionTypes from './actionTypes';
+import { AnyAction, combineReducers, Reducer } from 'redux';
+import type * as T from './types';
+import type { StoreState } from '../types';
+
+export const INITIAL_STATE: T.RafflesState = {
+  error: null,
+  isLoading: false,
+  result: null,
+  raffle: null,
+};
+
+const error = (state = INITIAL_STATE.error, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_RAFFLES_REQUEST:
+    case actionTypes.FETCH_RAFFLE_REQUEST:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_REQUEST:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_REQUEST:
+      return INITIAL_STATE.error;
+    case actionTypes.FETCH_RAFFLES_FAILURE:
+    case actionTypes.FETCH_RAFFLE_FAILURE:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_FAILURE:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_FAILURE:
+      return action.payload.error;
+    default:
+      return state;
+  }
+};
+
+const isLoading = (state = INITIAL_STATE.isLoading, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_RAFFLES_REQUEST:
+    case actionTypes.FETCH_RAFFLE_REQUEST:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_REQUEST:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_REQUEST:
+    case actionTypes.CREATE_RAFFLE_PARTICIPATION_REQUEST:
+    case actionTypes.MODIFY_RAFFLE_PARTICIPATION_REQUEST:
+      return true;
+    case actionTypes.FETCH_RAFFLES_SUCCESS:
+    case actionTypes.FETCH_RAFFLES_FAILURE:
+    case actionTypes.FETCH_RAFFLE_SUCCESS:
+    case actionTypes.FETCH_RAFFLE_FAILURE:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_SUCCESS:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_FAILURE:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_FAILURE:
+    case actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE:
+    case actionTypes.MODIFY_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.MODIFY_RAFFLE_PARTICIPATION_FAILURE:
+      return INITIAL_STATE.isLoading;
+    default:
+      return state;
+  }
+};
+
+const result = (state = INITIAL_STATE.result, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_RAFFLES_SUCCESS:
+      return action.payload.result;
+    default:
+      return state;
+  }
+};
+
+const raffle = (state = INITIAL_STATE.raffle, action: AnyAction) => {
+  if (action.type === actionTypes.FETCH_RAFFLE_SUCCESS) {
+    return action.payload.result;
+  }
+  return state;
+};
+
+const resetEntitiesStateReducer = (
+  state: NonNullable<StoreState['entities']>,
+) => {
+  const { raffles, ...rest } = state;
+
+  return rest;
+};
+
+export const entitiesMapper = {
+  [actionTypes.FETCH_RAFFLE_ESTIMATION_SUCCESS]: (
+    state: NonNullable<StoreState['entities']>,
+    action: AnyAction,
+  ) => {
+    const { raffleId, estimation } = action.payload;
+    if (!estimation) {
+      return state;
+    }
+    return {
```

Comment:
This may be useful to you, but consider using [immer](https://github.com/immerjs/immer) here, ([example](https://github.com/Farfetch/blackout/blob/next/packages/redux/src/users/addresses/reducer.ts#L171))

Reply:
I already change it...can you double check ?